ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/integration/RestApiTest.java

```diff
@@ -879,7 +898,103 @@ public void shouldExecutePullQueryOverRest() {
+    }
+
+    @Test
+    public void shouldExecutePullQueryOverRestWithBound() {
+      // Given:
+      final String clientCV = CONSISTENCY_OFFSET_VECTOR_BEFORE.serialize();
+      final String serverCV = CONSISTENCY_OFFSET_VECTOR.serialize();
+      final ImmutableMap.Builder<String, Object> builder = new ImmutableMap.Builder<String, Object>()
+          .put(KSQL_QUERY_PULL_CONSISTENCY_OFFSET_VECTOR_ENABLED, true)
+          .put(KsqlRequestConfig.KSQL_REQUEST_QUERY_PULL_CONSISTENCY_OFFSET_VECTOR, clientCV);
+      final Map<String, Object> requestProperties = builder.build();
+      final Supplier<List<String>> call = () -> {
+        final String response = rawRestQueryRequest(
+            "SELECT COUNT, USERID from " + AGG_TABLE + " WHERE USERID='" + AN_AGG_KEY + "';",
+            MediaType.APPLICATION_JSON,
+            Collections.emptyMap(),
+            requestProperties
+        );
+        return Arrays.asList(response.split(System.lineSeparator()));
+      };
+
+
+      // When:
+      final List<String> messages = assertThatEventually(call, hasSize(HEADER + 2));
+
+      System.out.println("messages = " + messages);
+      // Then:
+      assertThat(messages, hasSize(HEADER + 2));
+      assertThat(messages.get(0), startsWith("[{\"header\":{\"queryId\":\""));
+      assertThat(messages.get(0),
+                 endsWith("\",\"schema\":\"`COUNT` BIGINT, `USERID` STRING KEY\"}},"));
+      assertThat(messages.get(1), is("{\"row\":{\"columns\":[1,\"USER_1\"]}},"));
+      assertThat(messages.get(2), is("{\"consistencyToken\":{\"consistencyToken\":"
+                                         + "\"" + serverCV + "\"}}]"));
+      verifyConsistencyVector(messages.get(2), CONSISTENCY_OFFSET_VECTOR);
+    }
+
+    @Test
+    public void shouldFailPullQueryOverRestWithBound() {
+      // Given:
+      final String serializedCV = CONSISTENCY_OFFSET_VECTOR_AFTER.serialize();
+      final ImmutableMap.Builder<String, Object> builder = new ImmutableMap.Builder<String, Object>()
+          .put(KSQL_QUERY_PULL_CONSISTENCY_OFFSET_VECTOR_ENABLED, true)
+          .put(KsqlRequestConfig.KSQL_REQUEST_QUERY_PULL_CONSISTENCY_OFFSET_VECTOR, serializedCV);
+      final Map<String, Object> requestProperties = builder.build();
+
+      // When:
+      final Supplier<List<String>> call = () -> {
+        final String response = rawRestQueryRequest(
+            "SELECT COUNT, USERID from " + AGG_TABLE + " WHERE USERID='" + AN_AGG_KEY + "';",
+            MediaType.APPLICATION_JSON,
+            Collections.emptyMap(),
+            requestProperties
+        );
+        return Arrays.asList(response.split(System.lineSeparator()));
+      };
+
+      // Then:
+      final List<String> messages = assertThatEventually(call, hasSize(HEADER + 1));
+      System.out.println("messages = " + messages);
+      assertThat(messages, hasSize(HEADER + 1));
+      assertThat(messages.get(1), containsString("Error in processing query."));
```

Comment:
We currently don't return any meaningful error code or message, hence the only thing I can do is check for the presence of AN error